Reduce the number of calculating usage

FengyunPan opened the pull request:
The quotaEvaluator call Usage() to calculate usage twice for every
admission attributes, and calculating usage will waste much time.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/44149)
<!-- Reviewable:end -->

FengyunPan commented:
/release-note-none

derekwaynecarr reviewed (changes requested):
if we move the code as noted, i think we need to make sure the block here is also moved above.

https://github.com/kubernetes/kubernetes/pull/44149/files#diff-1622c871bcd912b43639c9d3d441e713L445

Commit:
Reduce the number of calculating usage

The quotaEvaluator call Usage() to calculate usage twice for every
admission attributes, and calculating usage will waste much time.
Let's reduce it.

FengyunPan commented:
@derekwaynecarr Alright, I will update it soon.

FengyunPan commented:
@derekwaynecarr Thank for your review, I have updated it.

derekwaynecarr commented:
@k8s-bot ok to test

derekwaynecarr commented on plugin/pkg/admission/resourcequota/controller.go:
```diff
@@ -379,17 +379,28 @@ func (e *quotaEvaluator) checkRequest(quotas []api.ResourceQuota, a admission.At
+		}
+	}
+
+	// if we have limited resources enabled for this resource, always calculate usage
```
this issue i have with the code now is that usage is calculated all the time.  we should only need to calculate use if there is a matching limited resource (which is an alpha feature at moment) or if there is an actually interesting covering quota.  can this be structured to ensure that usage is only calculated if the above constraints are satisfied?

derekwaynecarr reviewed (changes requested):
this appears to calculate usage even if there is no interesting quota or limited resource.  we should only calculate usage if there is an interesting quota or a limited resource.
